packages/devtools_app/lib/src/memory/flutter/memory_screen.dart

```diff
@@ -150,82 +219,127 @@ class MemoryBodyState extends State<MemoryBody> {
 */
   }
 
+  static const minIncludeTextLeftButtons = 1300;
+  static const minIncludeTextRightButtons = 1100;
+
   Widget _leftsideButtons() {
-    return Row(
-      crossAxisAlignment: CrossAxisAlignment.center,
-      children: [
-        OutlineButton(
-          key: MemoryScreen.pauseButtonKey,
-          onPressed: controller.paused ? null : _pauseLiveTimeline,
-          child: const MaterialIconLabel(
-            Icons.pause,
-            'Pause',
-            minIncludeTextWidth: 900,
+    return Container(
+      padding: const EdgeInsets.symmetric(horizontal: 6),
+      // Add a semi-transparent background to the
+      // expand and collapse buttons so they don't interfere
+      // too badly with the tree content when the tree
+      // is narrow.
+      color: Theme.of(context).scaffoldBackgroundColor.withAlpha(200),
+      child: Row(
+        crossAxisAlignment: CrossAxisAlignment.center,
+        children: [
+          OutlineButton(
+            key: MemoryScreen.pauseButtonKey,
+            onPressed: controller.paused ? null : _pauseLiveTimeline,
+            child: Label(
+              pauseIcon,
+              'Pause',
+              minIncludeTextWidth: 1300,
+            ),
           ),
-        ),
-        OutlineButton(
-          key: MemoryScreen.resumeButtonKey,
-          onPressed: controller.paused ? _resumeLiveTimeline : null,
-          child: const MaterialIconLabel(
-            Icons.play_arrow,
-            'Resume',
-            minIncludeTextWidth: 900,
+          OutlineButton(
+            key: MemoryScreen.resumeButtonKey,
+            onPressed: controller.paused ? _resumeLiveTimeline : null,
+            child: Label(
+              playIcon,
+              'Resume',
+              minIncludeTextWidth: 1300,
+            ),
           ),
-        ),
-      ],
+          const SizedBox(width: 16.0),
+          OutlineButton(
+              key: MemoryScreen.clearButtonKey,
+              onPressed: controller.memorySource == MemoryController.liveFeed
+                  ? _clearTimeline
+                  : null,
+              child: Label(
+                clearIcon,
+                'Clear',
+                minIncludeTextWidth: 1300,
+              )),
```

Comment:
trailing comma

Reply:
@pq 
Would it violate the rules for writing lints if we had a lint that depended on how the code was formatted?
It would be great to have a lint that complained anytime you had code like `))` and the matching open parens were not on the same line as the `))`.
 within a build method and quick fixed to add a trailing comma.
This would be a little odd as it is a lint that could be triggered or go away due to running the formatter which might change whether the `))` were on the same line or different lines.

Reply:
I do think that in general we probably don't want to flag things that will get formatted away but I'm not sure.  There's some precedent in [lines_longer_than_80_chars](http://dart-lang.github.io/linter/lints/lines_longer_than_80_chars.html) but I don't know how much adoption it's seen.  There's also this open request which has a similar vibe https://github.com/dart-lang/linter/issues/1338.

Reply:
yeah. you really want a be able to ensure this lint only runs if the file is already correctly formatted by dartfmt.

Reply:
Done.